.github/workflows/main_cd.yml

```diff
@@ -37,35 +37,88 @@ jobs:
 
       # 디렉토리 생성
       - name: Make Directory
-        run: mkdir -p football
+        run: mkdir -p football football/websocket football/api
 
       # Api 서버 Jar 파일 복사
       - name: Copy Jar
-        run: cp ./football-api-server/build/libs/*.jar ./football
+        run: cp ./football-api-server/build/libs/*.jar ./football/api
 
       # Websocket Jar 파일 복사
       - name: Copy Jar
-        run: cp ./football-websocket-server/build/libs/*.jar ./football
+        run: cp ./football-websocket-server/build/libs/*.jar ./football/websocket
 
       # 웹소켓 서버(= 배스천 서버)에 생성된 jar 파일이 저장된 폴더를 전송
-      - name: Deliver File
+      - name: Deliver File To Bastion Server
         uses: appleboy/scp-action@master
         with:
           host: ${{ secrets.SSH_WC_HOST }} # 인스턴스 IP 주소
           username: ${{ secrets.SSH_WC_USERNAME }} # 인스턴스에 지정된 user name(ex. ubuntu, ec2-user 등)
           key: ${{ secrets.SSH_KEY }} # SSH key의 private key
           port: ${{ secrets.SSH_PORT }} # port 번호(보통 22 포트 사용)
-          source: "./football"
+          source: "./"
           target: "source"
           rm: true
 
-      # 배스천 서버에서 api 서버로 jar 파일만 따로 전송
-      - name: Deploy
+      # 배스천 서버에 접속해 api 서버로 jar 파일만 따로 전송
+      - name: Deliver File To Private Server
         uses: appleboy/ssh-action@master
         with:
           host: ${{ secrets.SSH_WC_HOST }}
           username: ${{ secrets.SSH_WC_USERNAME }}
           key: ${{ secrets.SSH_KEY }}
           port: ${{ secrets.SSH_PORT }}
           script:
-            scp -i ./.ssh/shkim-keypair.cer ./source/football/football-api-server-1.0-SNAPSHOT.jar ${{ secrets.SSH_API_USERNAME }}@${{ secrets.SSH_API_HOST }}:./source/football
+            SOURCE_DIR=./source/football/api
+            FILE_NAME=`find $SOURCE_DIR/*.jar -printf "%f\n"`
+            
+            scp -i ./.ssh/shkim-keypair.cer $SOURCE_DIR/$FILE_NAME ${{ secrets.SSH_API_USERNAME }}@${{ secrets.SSH_API_HOST }}:$SOURCE_DIR
+
+      # Api 서버 실행
+      - name: Run Api Server
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.SSH_WC_HOST }}
+          username: ${{ secrets.SSH_WC_USERNAME }}
+          key: ${{ secrets.SSH_KEY }}
+          port: ${{ secrets.SSH_PORT }}
+          script: |
+            ssh -i ./.ssh/shkim-keypair.cer ${{ secrets.SSH_API_USERNAME }}@${{ secrets.SSH_API_HOST }}
```

Comment:
멘토님 private 서버인 api 서버에 접속하기 위해 배스천 서버 역할을 하는 websocket 서버에서 ssh 명령어로 api 서버로 접속하도록 커맨드를 구성해봤는데 제대로 동작하지 않는 것 같습니다! 

오히려 ssh 명령어만 동작하지 않고 웹소켓 서버에서 api  jar 파일과 websocket jar 파일이 실행되고 있어 workflow는 잘 동작하는 것처럼 보입니다ㅠㅠ

private 서버에 접속해 jar 파일을 실행시킬려면 어떤 부분을 수정해봐야 할지 궁금합니다!

Reply:
해당 job 수행시 메세지는 다음과 같습니다!
```
======CMD======
ssh -i ./.ssh/shkim-keypair.cer ***@***

SOURCE_DIR=./source/football/api
FILE_NAME=`find $SOURCE_DIR/*.jar -printf "%f\n"`
PID=`ps -ef | grep occupying | grep sudo | grep -v "bash -c" | awk '{print $2}'`

if [ -z "$PID" ]; then
echo "#### THERE IS NO PROCESS ####"
else
echo "#### KILL $PID ####"
sudo kill $PID
fi

echo "#### RUN $SOURCE_DIR/$FILE_NAME ####"

sudo java -jar -Dspring.profiles.active=prod $SOURCE_DIR/$FILE_NAME > /dev/null 2>&1 &

======END======
err: Pseudo-terminal will not be allocated because stdin is not a terminal.
out: Welcome to Ubuntu ***.04.1 LTS (GNU/Linux 5.15.0-1019-aws x86_64)
out: 
out:  * Documentation:  https://help.***.com
out:  * Management:     https://landscape.canonical.com
out:  * Support:        https://***.com/advantage
out: 
out:   System information as of Sun Sep 11 06:20:38 UTC 20***
out: 
out:   System load:  0.17724609[37](https://github.com/f-lab-edu/football/runs/8290196973?check_suite_focus=true#step:13:38)5     Processes:             100
out:   Usage of /:   [49](https://github.com/f-lab-edu/football/runs/8290196973?check_suite_focus=true#step:13:50).6% of 7.[57](https://github.com/f-lab-edu/football/runs/8290196973?check_suite_focus=true#step:13:58)GB   Users logged in:       1
out:   Memory usage: [58](https://github.com/f-lab-edu/football/runs/8290196973?check_suite_focus=true#step:13:59)%               IPv4 address for eth0: ***
out:   Swap usage:   0%
out: 
out: 
out: 3 updates can be applied immediately.
out: To see these additional updates run: apt list --upgradable
out: 
out: 
out: The list of available updates is more than a week old.
out: To check for new updates run: sudo apt update
out: Failed to connect to https://changelogs.***.com/meta-release-lts. Check your Internet connection or proxy settings
out: 
out: 
out: #### THERE IS NO PROCESS ####
out: #### RUN ./source/football/api/football-api-server-1.0-SNAPSHOT.jar ####
```

Reply:
베스천을 통해서 다른 서버에 접속한 뒤 커맨드가 실행되어야 하는데, 연결된 이후에 커맨드도 베스천에서 실행되고 있나보네요.
깃헙 액션에서 이런 형태를 어떻게 구성하는지 한 번 찾아보시는게 도움이 될 것 같네요.
혹시 찾아보신게 있으실까요.?

Reply:
네..배포 후 확인해보면 배스천 서버에 두 서버가 같이 떠져있습니다ㅠㅠ 
마땅히 참고할만한 자료는 찾지 못한 상황입니다!😭